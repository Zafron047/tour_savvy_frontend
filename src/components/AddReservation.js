import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addReservation } from '../redux/reservations/reservationSlice';
import '../stylesheets/addReservation.css';
import '../stylesheets/common.css';

function AddReservation() {
  const dispatch = useDispatch();
  const packages = useSelector((state) => state.packages.allPackages);

  const [cityName, setCityName] = useState('');
  const [reservationDate, setReservationDate] = useState('');
  const [packageName, setPackageName] = useState('');
  const [packageType, setPackageType] = useState('');
  const [packageId, setPackageId] = useState(0);

  let packageTypes = [];

  packages.forEach((packageItem) => {
    if (packageItem.id === packageId) {
      packageTypes = packageItem.package_type;
    }
  });

  const user = JSON.parse(localStorage.getItem('user'));
  if (!user) {
    return <div>Please log in to add a reservation.</div>;
  }
  const handleSubmit = async (e) => {
    e.preventDefault();
    dispatch(
      addReservation({
        cityName,
        reservationDate,
        packageName,
        packageType,
      }),
    );
  };

  return (
    <div className="addReservation">
      <div className="background" />
      <div className="main-addReservation">
        <div className="addReservation-header">
          <h2>BOOK A RESERVATION WITH US</h2>
          <p>
            Select from a wide range of locations and packages to suit your needs
            and enjoy your holidays and vacations with us, we guarantee your
            satisfaction on all reservatoins made with us.
          </p>
        </div>
        <form onSubmit={handleSubmit} className="forms">
          <div>
            <input
              type="text"
              className="form-fields"
              placeholder="City Name"
              value={cityName}
              onChange={(e) => setCityName(e.target.value)}
            />
          </div>

          <div>
            <input
              type="date"
              className="form-fields"
              placeholder="Reservation Date"
              value={reservationDate}
              onChange={(e) => setReservationDate(e.target.value)}
            />
          </div>
          <div>
            <select
              className="form-fields"
              value={packageName}
              onChange={(e) => {
                const selectedPackage = packages.find(
                  (packageItem) => packageItem.name === e.target.value,
                );
                if (selectedPackage) {
                  setPackageName(e.target.value);
                  setPackageId(selectedPackage.id);
                }
              }}
            >
              <option value="" className="form-fields">
                Select a Package
              </option>
              {packages.map((packageItem) => (
                <option
                  key={packageItem.id}
                  value={packageItem.name}
                  className="form-fields"
                >
                  {packageItem.name}
                </option>
              ))}
            </select>
          </div>
          <div>
            {packageTypes.map((type) => (
              <div key={type.name} className="package-types">
                <input
                  type="radio"
                  name="packageType"
                  value={type.name}
                  checked={packageType === type.name}
                  onChange={(e) => setPackageType(e.target.value)}
                />
                <label htmlFor="PackageType" className="radio">{type.name}</label>
              </div>
            ))}
          </div>

          <button type="submit" className="form-btns btn">
            Add Reservation
          </button>
        </form>
      </div>
    </div>
  );
}
export default AddReservation;
