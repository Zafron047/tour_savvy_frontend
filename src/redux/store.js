import { configureStore } from '@reduxjs/toolkit';
import packagesSlice from './packages/packagesSlice';
import reservationsReducer from './reservations/reservationSlice';
import userSlice from './authentication/userSlice';

const loadStateFromLocalStorage = () => {
  try {
    const serializedState = localStorage.getItem('reduxState');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    return undefined;
  }
};

const loadUserFromLocalStorage = () => {
  try {
    const serializedUser = localStorage.getItem('user');
    if (serializedUser === null) {
      return undefined;
    }
    return JSON.parse(serializedUser);
  } catch (err) {
    return undefined;
  }
};

const store = configureStore({
  reducer: {
    packages: packagesSlice,
    reservations: reservationsReducer,
    current_user: userSlice,
  },
  preloadedState: {
    current_user: loadUserFromLocalStorage(),
    ...loadStateFromLocalStorage(),
  },
  reservations: reservationsReducer,
});

export default store;
