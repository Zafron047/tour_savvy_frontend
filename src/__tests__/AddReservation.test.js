import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import AddReservation from '../components/AddReservation';

test('Should render AddReservation component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <AddReservation />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
