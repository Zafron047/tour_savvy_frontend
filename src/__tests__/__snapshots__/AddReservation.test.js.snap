// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AddReservation component correctly 1`] = `
<div>
  <div>
    Please log in to add a reservation.
  </div>
</div>
`;
